lib/import-export-visitor.js

```diff
@@ -303,15 +318,16 @@ IEVp.finalizeHoisting = function () {
     delete bodyInfo.hoistedExportsMap;
     delete bodyInfo.hoistedExportsString;
     delete bodyInfo.hoistedImportsString;
-  }, this);
+  }
 
   // Just in case we call finalizeHoisting again, don't hoist anything.
   this.bodyInfos.length = 0;
 
-  this.removals.forEach(processRemoval);
+  let removalCount = this.removals.length;
 
-  // Just in case we call finalizeHoisting again, don't remove anything.
-  this.removals.length = 0;
+  while (removalCount--) {
+    processRemoval(this.removals.pop());
```

Comment:
I don't think this is a hot spot in the code.

With a top-level parse, this loop happens once per file (because there's only one scope where we hoist anything), and the length of `this.removals` is the number of `import` and `export` declarations we're replacing. Setting `this.removals.length = 0` once per file is never going to be a performance bottleneck, and if it was, we could just as easily not modify the array at all (no `.pop()`, no `.length = 0`, just the loop).

Given that, I would prefer shorter code (`this.removals.forEach(processRemoval)`).